packages/@aws-cdk/aws-appmesh/lib/mesh.ts

```diff
@@ -40,6 +40,15 @@ export interface IMesh extends cdk.IResource {
    */
   readonly meshArn: string;
 
+  /**
+   * The AWS IAM account ID of the AppMesh mesh
+   *
+   * @see https://docs.aws.amazon.com/app-mesh/latest/userguide/sharing.html
+   *
+   * @attribute
+   */
+  readonly meshOwner: string;
```

Comment:
Hi @skinny85, 

Just picking up this PR from Seiya.


What you're trying to say here is that rather than having meshOwner as a property to the `IMesh`, we should use the `env.account` property that is already set (or will be, once we make the below change)? 

However, based on what I understand, won't that be a problem when in a shared mesh case, the meshOwner is somebody other than that account (which will be the account stored in `env.account`?) What I mean is a case like [this](https://github.com/aws/aws-app-mesh-examples/blob/master/walkthroughs/howto-cross-account/secondary-account/mesh.yaml) where the Virtual Node has a [different mesh owner](https://github.com/aws/aws-app-mesh-examples/blob/master/walkthroughs/howto-cross-account/secondary-account/deploy.sh#L42).

Reply:
Not sure I understand. Are you saying a VirtualNode can have a different MeshOwner than the Mesh it belongs to?

Reply:
Yes, that's the use case that we are trying to support by adding meshOwner, in order to create the ability for customers to have meshes that can be shared across accounts. The links in the above comment show a virtual node being created in a separate account

Reply:
1) The env.account that will be populated will be the account of the customer creating the resource? 

2) Or will it be the account of the customer who created the mesh? 

If 2's the case then we don't need this field, but if it is 1 then it will be required so they can specify a different account Id at the time of resource creation.

Reply:
Sorry, I don't understand what you're saying.

Can you answer my question above?

Reply:
`env.account` means the AWS account that the Mesh belongs to.

Reply:
Discussed this separately, we can pull in the account of the meshOwner from the mesh ARN is CDK so we have decided against exposing this as a field.